extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/owserver/OwserverConnection.java

```diff
@@ -0,0 +1,452 @@
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+
+@NonNullByDefault
+public class OwserverConnection {
+    public static final String DEFAULT_HOST = "localhost";
+    public static final int DEFAULT_PORT = 4304;
+    public static final int KEEPALIVE_INTERVAL = 1000;
+
+    private static final int CONNECTION_MAX_RETRY = 5;
+
+    private final Logger logger = LoggerFactory.getLogger(OwserverConnection.class);
+
+    private final OwserverBridgeHandler thingHandlerCallback;
+    private String owserverAddress = DEFAULT_HOST;
+    private int owserverPort = DEFAULT_PORT;
+
+    private @Nullable Socket owserverSocket = null;
+    private @Nullable DataInputStream owserverInputStream = null;
+    private @Nullable DataOutputStream owserverOutputStream = null;
+    private OwserverConnectionState owserverConnectionState = OwserverConnectionState.STOPPED;
+
+    // reset to 0 after successful request
+    private int connectionErrorCounter = 0;
+
+    public OwserverConnection(OwserverBridgeHandler thingHandlerCallback) {
+        this.thingHandlerCallback = thingHandlerCallback;
+    }
+
+    /**
+     * set the owserver host address
+     *
+     * @param address as String (IP or FQDN), defaults to localhost
+     */
+    public void setHost(String address) {
+        this.owserverAddress = address;
+        if (owserverConnectionState != OwserverConnectionState.STOPPED) {
+            close();
+        }
+    }
+
+    /**
+     * set the owserver port
+     *
+     * @param port defaults to 4304
+     */
+    public void setPort(int port) {
+        this.owserverPort = port;
+        if (owserverConnectionState != OwserverConnectionState.STOPPED) {
+            close();
+        }
+    }
+
+    /**
+     * start the owserver connection
+     */
+    public void start() {
+        connectionErrorCounter = 0;
+        owserverConnectionState = OwserverConnectionState.CLOSED;
+        boolean success = false;
+        do {
+            success = open();
+        } while (success != true && owserverConnectionState != OwserverConnectionState.FAILED);
+    }
+
+    /**
+     * stop the owserver connection
+     */
+    public void stop() {
+        close();
+        owserverConnectionState = OwserverConnectionState.STOPPED;
+        thingHandlerCallback.reportConnectionState(owserverConnectionState);
+    }
+
+    /**
+     * list all devices on this owserver
+     *
+     * @return a list of device ids
+     */
+    public List<String> getDirectory() throws OwException {
+        List<String> directory = new ArrayList<String>();
+        OwserverPacket requestPacket = new OwserverPacket(OwserverMessageType.DIR, "/");
+
+        write(requestPacket);
+
+        OwserverPacket returnPacket = null;
+        do {
+            try {
+                returnPacket = read(false);
+            } catch (OwException e) {
+                logger.info("getDirectory may have returned incomplete result: {}", e.getMessage());
```

Comment:
Is that a software issue? Then better do WARN and include the exception as a last param.

Reply:
reduced to DEBUG, this migth happen in case of issues with the network or busg in the owserver, not in the binding.